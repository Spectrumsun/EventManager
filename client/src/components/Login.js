import React from 'react';

const styles = {
  width: '30rem'
};

const login = () => (
  <div>
    <div className="container">
      <div className="card loginCard" style={styles}>
        <div className="card-header">
          <h3>Login</h3>
        </div>
        <div className="card-body">
          <div className="cont card-body">
            <form action="viewevent.html" className="centerform">
              <div className="form-group">
                <label htmlFor="exampleInputEmail1">Email address</label>
                <input type="email" className="form-control form-control-lg" id="exampleInputEmail1" aria-describedby="emailHelp" placeholder="" required />
              </div>
              <div className="form-group">
                <label htmlFor="exampleInputPassword1">Password</label>
                <input type="password" className="form-control form-control-lg" id="exampleInputPassword1" placeholder required />
                <small ><a href="passwordreset.html" className="center-item">Password Reset</a></small>
                <br />
              </div>
              <div className="text-center">
                <button type="submit" className="btn btn-outline-dark">Submit</button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default login;
