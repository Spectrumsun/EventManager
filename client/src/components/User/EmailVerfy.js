import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as action from '../../store/actions/index';

class Login extends Component {
  componentWillMount() {
    this.props.initemailverify(this.props.match.params.token, this.props.history);
  }

  render() {
    return (
      <div />
    );
  }
}

const mapDispatchToProps = dispatch => ({
  initemailverify: (token, history) => dispatch(action.initemailverify(token, history)),
});

export default connect(null, mapDispatchToProps)(Login);
