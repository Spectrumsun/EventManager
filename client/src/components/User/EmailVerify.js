import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'; 
import * as action from '../../store/actions/index';

class EmailVerify extends Component {
  componentWillMount() {
    this.props.initemailverify(this.props.match.params.token, this.props.history);
  }

  render() {
    const isLoading = (
      <div className="loader" />
    );

    return (
      <div>
        <h2>Verifying Email</h2>
        { isLoading }
      </div>
    );
  }
}


EmailVerify.propTypes = {
  initemailverify: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      token: PropTypes.string,
    }),
  }).isRequired,
};

const mapDispatchToProps = dispatch => ({
  initemailverify: (token, history) => dispatch(action.initemailverify(token, history)),
});

export default connect(null, mapDispatchToProps)(EmailVerify);
