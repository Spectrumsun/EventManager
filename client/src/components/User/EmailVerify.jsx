import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as action from '../../store/actions/index';

/**
 * @class EmailVerify
 *
 * @extends {React.Component}
 */
export class EmailVerify extends Component {
  /**
   * @description run action on component mount to reload data
   *
   * @param {any} props.params.token
   *
export class EmailVerify extends Component {
   * @memberof Signup
   */
  componentWillMount() {
    this.props.initemailverify(
      this.props.match.params.token,
      this.props.history
    );
  }


  /**
   * @description renders component to the DOM
   *
   * @memberof EmailVerify
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    const isLoading = (
      <div className="loader" />
    );

    return (
      <div>
        <h2>Verifying Email</h2>
        { isLoading }
      </div>
    );
  }
}


EmailVerify.propTypes = {
  initemailverify: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      token: PropTypes.string,
    }),
  }).isRequired,
};

const mapDispatchToProps = dispatch => ({
  initemailverify: (token, history) =>
    dispatch(action.initemailverify(token, history)),
});

export default connect(null, mapDispatchToProps)(EmailVerify);
