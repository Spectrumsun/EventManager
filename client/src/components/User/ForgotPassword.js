import React, { Component } from 'react';
import { connect } from 'react-redux';
import toast from 'toastr';
import PropTypes from 'prop-types';
import TextField from './TextField';
import * as action from '../../store/actions/index';

class ForgotPassword extends Component {
  state = {
    email: '',
  }

  onChange =(e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit = (e) => {
    e.preventDefault();
    if (this.state.email === '') {
      toast.error('Email cannot be blank');
    } else {
      this.props.initconfirmPassword(this.state, this.props.history);
    }
  }

  render() {
    return (
      <div>
        <div className="container" style={{ paddingTop: '100px' }}>
          <div className="card loginCard" style={{ width: '30rem' }}>
            <div className="card-header">
              <h3>Forgot Password</h3>
            </div>
            <div className="card-body">
              <div className="cont card-body">
                <form onSubmit={this.onSubmit} className="centerform">
                  <TextField
                    label="Email"
                    value={this.state.email}
                    onChange={this.onChange}
                    name="email"
                    type="email"
                    placeholder="Vaild Email"
                  />
                  <div className="text-center">
                    <button type="submit" className="btn btn-outline-dark">Submit</button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

ForgotPassword.propTypes = {
  initconfirmPassword: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired
};

const mapDispatchToProps = dispatch => ({
  initconfirmPassword: (user, history) => dispatch(action.initconfirmPassword(user, history)),
});

export default connect(null, mapDispatchToProps)(ForgotPassword);
