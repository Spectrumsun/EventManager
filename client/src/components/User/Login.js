import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import toast from 'toastr';
import { connect } from 'react-redux';
import TextField from './TextField';
import * as action from '../../store/actions/index';

class Login extends Component {
  state = {
    email: '',
    password: ''
  }

  onChange =(e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit = (e) => {
    e.preventDefault();
    this.props.initUserLogin(this.state, this.props.history);
  }

  render() {
    return (
      <div>
        <div className="container" style={{ paddingTop: '100px' }}>
          <div className="card loginCard" style={{ width: '30rem' }}>
            <div className="card-header">
              <h3>Login</h3>
            </div>
            <div className="card-body">
              <div className="cont card-body">
                <form onSubmit={this.onSubmit} className="centerform">
                  <TextField
                    label="Email"
                    value={this.state.email}
                    onChange={this.onChange}
                    name="email"
                    type="email"
                    placeholder="Vaild Email"
                  />

                  <TextField
                    label="Password"
                    value={this.state.password}
                    onChange={this.onChange}
                    name="password"
                    type="password"
                    placeholder="Password"
                  />
                  <small ><Link to="/forgotpassword" className="center-item">Password Reset</Link></small>

                  <div className="text-center">
                    <button type="submit" className="btn btn-outline-dark">Submit</button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}



const mapDispatchToProps = dispatch => ({
  initUserLogin: (user, history) => dispatch(action.initUserLogin(user, history)),
});

export default connect(null, mapDispatchToProps)(Login);
