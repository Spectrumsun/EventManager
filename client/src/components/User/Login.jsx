import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import toast from 'toastr';
import TextField from './TextField';
import * as action from '../../store/actions/index';

/**
 * @class Login
 *
 * @extends {React.Component}
 */
export class Login extends Component {
  state = {
    email: '',
    password: '',
    formValid: false
  }

  /**
   * @description update component state with current value in dom
   *
   * @param {any} event
   *
   * @memberof Login
   *
   * @returns {void}
   */
  onChange =(event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @description vaildate data in state
   * sends state to api with action dispatch
   * @param {any} event
   *
   * @memberof Login
   *
   * @returns {void}
   */
  onSubmit = (event) => {
    event.preventDefault();
    this.setState({ formValid: false });
    if (this.state.email === '') {
      toast.error('Email cannot be blank');
    } else if (this.state.password === '') {
      toast.error('Password cannot be blank');
    } else {
      this.setState({ formValid: true });
      this.props.initUserLogin(
        this.state,
        this.props.history
      ).then(() => {
        this.setState({ formValid: false });
      });
    }
  }

  /**
   * @description renders component to the DOM
   *
   * @memberof ForgotPassword
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    return (
      <div>
        <div className="container" style={{ paddingTop: '150px' }}>
          <h3 className="center">Event Manager</h3>
          <div className="card loginCard" style={{ width: '30rem' }}>
            <div className="card-header">
              <h3>Login</h3>
            </div>
            <div className="card-body">
              <div className="cont card-body">
                <form onSubmit={this.onSubmit} className="centerform">
                  <TextField
                    label="Email"
                    value={this.state.email}
                    onChange={this.onChange}
                    name="email"
                    type="email"
                    placeholder="Vaild Email"
                  />

                  <TextField
                    label="Password"
                    value={this.state.password}
                    onChange={this.onChange}
                    name="password"
                    type="password"
                    placeholder="Password"
                  />
                  <small ><Link
                    to="/forgotpassword"
                    className="center-item"
                  >Forgot password?
                  </Link>
                  </small>

                  <div className="text-center">
                    <button
                      id="login"
                      type="submit"
                      className="btn btn-outline-dark"
                      disabled={this.state.formValid}
                    >Submit
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  initUserLogin: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired
};


const mapDispatchToProps = dispatch => ({
  initUserLogin: (user, history) =>
    dispatch(action.initUserLogin(user, history)),
});

export default connect(null, mapDispatchToProps)(Login);
