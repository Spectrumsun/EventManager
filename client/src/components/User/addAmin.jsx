import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import toast from 'toastr';
import TextField from './TextField';
import * as action from '../../store/actions/index';

/**
 * @class Login
 *
 * @extends {React.Component}
 */
export class AddAmin extends Component {
  state = {
    email: '',
    role: '',
  }

  /**
   * @description update component state with current value in dom
   *
   * @param {any} event
   *
   * @memberof Login
   *
   * @returns {void}
   */
  onChange =(e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  /**
   * @description vaildate data in state
   * sends state to api with action dispatch
   * @param {any} event
   *
   * @memberof Login
   *
   * @returns {void}
   */
  onSubmit = (e) => {
    e.preventDefault();
    if (this.state.email === '') {
      toast.error('Email cannot be blank');
    } else if (this.state.role === '') {
      toast.error('role cannot be blank');
    } else {
      this.props.initaddAdmin(
        this.state,
        this.props.history
      );
    }
  }

  /**
   * @description renders component to the DOM
   *
   * @memberof ForgotPassword
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    return (
      <div>
        <div className="container" style={{ paddingTop: '150px' }}>
          <h3 className="center">Event Manager</h3>
          <div className="card loginCard" style={{ width: '30rem' }}>
            <div className="card-header">
              <h3>Add New Admin</h3>
            </div>
            <div className="card-body">
              <div className="cont card-body">
                <form onSubmit={this.onSubmit} className="centerform">
                  <TextField
                    label="Email"
                    value={this.state.email}
                    onChange={this.onChange}
                    name="email"
                    type="email"
                    placeholder="Vaild Email"
                  />

                  <TextField
                    label="Role"
                    value={this.state.role}
                    onChange={this.onChange}
                    name="role"
                    type="text"
                    placeholder="role"
                  />
                  <div className="text-center">
                    <button
                      type="submit"
                      className="btn btn-outline-dark"
                    >Submit
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

AddAmin.propTypes = {
  initaddAdmin: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired
};



const mapDispatchToProps = dispatch => ({
  initaddAdmin: (state, history) =>
    dispatch(action.initaddAdmin(state, history)),
});


export default connect(mapDispatchToProps)(AddAmin);
