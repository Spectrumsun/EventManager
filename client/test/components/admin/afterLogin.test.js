import React from 'react';
import { shallow } from 'enzyme';
import Authenticate from '../../../src/components/Auth/afterLogin';
import { Login } from '../../../src/components/User/Login'

const components = Authenticate(Login);

const { WrappedComponent } = components;


let props;
let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<WrappedComponent {...props} />);
  }
  return mountedComponent;
};

describe('Component:  isAuthenticated User', () => {
  beforeEach(() => {
    props = {
      auth: {
        isAuthenticated: true,
        user: {
          id: 'jame',
          role: 'user'
        },
      },
      history: {
        push: jest.fn()
      }
    };
    mountedComponent = undefined;
  });

  describe('Protected Login Page after logging in test suite', () => {
    it('Redirects user to homepage if they access rhe login page after logging in', () => {
      props.isAuthenticated = true;
      expect(getComponent()).toMatchSnapshot();
    });
  });
});

