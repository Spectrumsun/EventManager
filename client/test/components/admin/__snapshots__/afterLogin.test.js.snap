// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  isAuthenticated User Protected Login Page after logging in test suite Redirects user to homepage if they access rhe login page after logging in 1`] = `
<Login
  auth={
    Object {
      "isAuthenticated": true,
      "user": Object {
        "id": "jame",
        "role": "user",
      },
    }
  }
  history={
    Object {
      "push": [MockFunction] {
        "calls": Array [
          Array [
            "/",
          ],
        ],
      },
    }
  }
  isAuthenticated={true}
/>
`;
